### ISSUE TITLE
PrecededBy: Fails when argument is not Literal - and infinite recursion if named

### ISSUE THREAD

**Issue by EdwardJB** [pyparsing/pyparsing#127]
Version `2.4.2`.

An unnamed `PrecededBy` with a `Literal` argument seems to work:

```python
string_test = 'notworking'
#negs = pyparsing.Or(['not', 'un'])('negs')
negs_pb = pyparsing.PrecededBy('not', retreat=100)#('negs_lb')
pattern = (negs_pb + pyparsing.Literal('working'))('main')

results = (list(pattern.scanString(string_test)) or [[None]])[0]
for result in results:
    if type(result) is pyparsing.ParseResults:
        print(result.dump())
    else:
        print(result)
```
```
['working']
- main: ['working']
```

But if I use a complex object instead of  `Literal`:

```python
string_test = 'notworking'
negs = pyparsing.Or(['not', 'un'])('negs')
negs_pb = pyparsing.PrecededBy(negs, retreat=100)('negs_lb')
pattern = (negs_pb + pyparsing.Literal('working'))('main')

results = (list(pattern.scanString(string_test)) or [[None]])[0]
for result in results:
    if type(result) is pyparsing.ParseResults:
        print(result.dump())
    else:
        print(result)
```
```
None
```

And if I use a `Literal` argument but name the `PrecededBy`:

```python
string_test = 'notworking'
#negs = pyparsing.Or(['not', 'un'])('negs')
negs_pb = pyparsing.PrecededBy('not', retreat=100)('negs_lb')
pattern = (negs_pb + pyparsing.Literal('working'))('main')

results = (list(pattern.scanString(string_test)) or [[None]])[0]
for result in results:
    if type(result) is pyparsing.ParseResults:
        print(result.dump())
    else:
        print(result)
```
```
  File "C:/repo/parsing/t_series.py", line 13, in <module>
    print(result.dump())
  File "C:\Python37\lib\site-packages\pyparsing.py", line 1094, in dump
    out.append(v.dump(indent=indent, full=full, include_list=include_list, _depth=_depth + 1))
  File "C:\Python37\lib\site-packages\pyparsing.py", line 1094, in dump
    out.append(v.dump(indent=indent, full=full, include_list=include_list, _depth=_depth + 1))
  File "C:\Python37\lib\site-packages\pyparsing.py", line 1094, in dump
    out.append(v.dump(indent=indent, full=full, include_list=include_list, _depth=_depth + 1))
  [Previous line repeated 985 more times]
  File "C:\Python37\lib\site-packages\pyparsing.py", line 1087, in dump
    items = sorted((str(k), v) for k, v in self.items())
  File "C:\Python37\lib\site-packages\pyparsing.py", line 1087, in <genexpr>
    items = sorted((str(k), v) for k, v in self.items())
  File "C:\Python37\lib\site-packages\pyparsing.py", line 654, in <genexpr>
    return ((k, self[k]) for k in self._iterkeys())
  File "C:\Python37\lib\site-packages\pyparsing.py", line 586, in __getitem__
    if isinstance(i, (int, slice)):
RecursionError: maximum recursion depth exceeded in __instancecheck__
```

**Comment by ptmcg** [comment]
Yes, pretty clearly a bug. Thanks for the sample code showing the problem.

**Comment by ptmcg** [comment]
I finally made some headway on this, I think I have a working solution, will post to GH shortly.

**Comment by ptmcg** [comment]
Fixed in 2.4.3, released today
